src/eigen/diag.jl

```diff
@@ -9,7 +9,8 @@ that really does the work, operating on a single ``k``-Block.
 function diagonalize_all_kblocks(eigensolver, ham::Hamiltonian, nev_per_kpoint::Int;
                                  ψguess=nothing,
                                  prec_type=PreconditionerTPA, interpolate_kpoints=true,
-                                 tol=1e-6, miniter=1, maxiter=100, n_conv_check=nothing)
+                                 tol=1e-6, miniter=1, maxiter=100, n_conv_check=nothing,
+                                 rng=default_rng())
```

Comment:
This is a very confusing name. I would expect `default_rng() = Random.default_rng()`, which is not the case. Call it `DFTK.default_rng()` the very least.